### PR TITLE
fix: fix redundant priv-validator-key generation when running with kms

### PR DESCRIPTION
**Pull request by jaeseung-bae**
## Description

* Fix for redundant key generation. With running kms, generating priv-key is unnecessary.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: .github/workflows/check-generated.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          make error-doc-gen
+          if ! git diff --stat --exit-code ; then
+            echo ">> ERROR:"
+            echo ">>"
+            echo ">> Error documents require update (source files in x folder may have changed)."
+            echo ">> Ensure your tools are up-to-date, re-run 'make error-doc' and update this PR."
+            echo ">>"
+            exit 1
+          fi
```
Comment (zemyblue) [inline .github/workflows/check-generated.yml:37]: Why does this file is added in this file?

Comment (jaeseung-bae) [reply .github/workflows/check-generated.yml]: This was added automatically when I click github ui to change base. I'll fix it.

**Review by zemyblue — COMMENTED** [review]
I think it's nice to add some explanation about doesn't generate a private key if some condition.

**Comment by jaeseung-bae** [comment]
> I think it's nice to add some explanation about doesn't generate a private key if some condition.

I named the function `genPvFileOnlyWhenKmsAddressEmpty` to express conditional generation of the key. 
The name doesn't seem to be enough. 
Do you have any idea?

**Comment by jaeseung-bae** [comment]
> > > I think it's nice to add some explanation about doesn't generate a private key if some condition.
> > 
> > 
> > I named the function `genPvFileOnlyWhenKmsAddressEmpty` to express conditional generation of the key. The name doesn't seem to be enough. Do you have any idea?
> 
> I think the condition of which does not generate a private key seems that the config.yml set `priv_validator_laddr` is empty string and there is no private key. Is it right?
> 
> If then, how about adding this explanation in config.yml. Oh, it is Ostracon's area. right?

If `priv_validator_laddr` is empty string, no key will be generated.
It's nice to put an explanation to config.yml. 
Yes, It's ostracon's area. I'll put it in ostracon's area.

**Review by zemyblue — DISMISSED** [review]

File: server/start_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"github.com/Finschia/ostracon/config"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestGenPvFileOnlyWhenKmsAddressEmptyGenerateFiles(t *testing.T) {
+	cfg := config.DefaultConfig()
+	cfg.PrivValidatorKey = "./key.json"
```
Comment (0Tech) [inline server/start_test.go:12]: How about using `os.MkdirTemp()`?

**Review by 0Tech — APPROVED** [review]

**Review by zemyblue — APPROVED** [review]

**Pull request merged**